elevation-plot/index.js

```diff
@@ -0,0 +1,109 @@
+import { createClient, createRequest, defaultExchanges, subscriptionExchange } from '@urql/core';
+import { pipe, subscribe } from 'wonka';
+import { SubscriptionClient } from 'subscriptions-transport-ws';
+import { createRoute, routeUpdate, getRoutePath } from './queries.js';
+import { drawRoute } from './map.js';
+import { loadGraph, imageLoader } from './graph';
+import * as mapboxPolyline from '@mapbox/polyline';
+import { displaySpecs, updateSpecs } from './journey-specs';
+
+/**
+ * For the purpose of this example we use urgl - lightweights GraphQL client.
+ * To establish a connection with Chargetrip GraphQL API you need to have an API key.
+ * Read more about an authorisation in our documentation (https://docs.chargetrip.com/#authorisation).
+ */
+const headers = {
+  'x-client-id': '5e8c22366f9c5f23ab0eff39',
+};
+
+const subscriptionClient = new SubscriptionClient('wss://api.chargetrip.io/graphql', {
+  reconnect: true,
+  connectionParams: headers,
+});
+
+const client = createClient({
+  url: 'https://api.chargetrip.io/graphql',
+  fetchOptions: {
+    method: 'POST',
+    headers,
+  },
+  exchanges: [
+    ...defaultExchanges,
+    subscriptionExchange({
+      forwardSubscription(operation) {
+        return subscriptionClient.request(operation);
+      },
+    }),
+  ],
+});
+
+/**
+ * To create a route you need to:
+ *
+ * 1. create a new route and receive back its ID;
+ * 2. subscribe to route updates in order to receive its details.
+ */
+client
+  .mutation(createRoute)
+  .toPromise()
+  .then(response => {
+    const routeId = response.data.newRoute;
+
+    const { unsubscribe } = pipe(
+      client.executeSubscription(createRequest(routeUpdate, { id: routeId })),
+      subscribe(result => {
+        const { status, route } = result.data.routeUpdatedById;
+
+        // you can keep listening to the route changes to update route information
+        // for this example we want to only draw the initial route
+        if (status === 'done' && route) {
+          unsubscribe();
+          const routeData = result.data.routeUpdatedById.route;
+          drawRoutePolyline(routeData, response.data.newRoute);
+          loadGraph(routeData, routeData.elevationPlot);
+          imageLoader(routeData, routeData.legs);
```

Comment:
this should be a part of the elevation graph functionality, no need to separate it and import it on its own. `loadGraph` uses route data as well, so legs information is presented there. Also if the first parameter is a routeData, then second `routeData.legs` doesn't make sense as you can always distract it ;)